pulp_rpm/app/access_policy.py

```diff
@@ -0,0 +1,39 @@
+from logging import getLogger
+
+from pulpcore.plugin.models import RepositoryVersion
+from pulpcore.plugin.access_policy import AccessPolicyFromDB
+
+from pulp_rpm.app.models.repository import RpmRepository
+
+_logger = getLogger(__name__)
+
+
+class CopyAccessPolicy(AccessPolicyFromDB):
+    """
+    Access policy for ContainerNamespaceViewSet.
+    """
+
+    def has_perms_to_copy(self, request, view, action):
+        """
+        Check if the source and destination repository matches the usernames permissions.
+        """
+        serializer = view.serializer_class(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+
+        dest_repo_id = request.data["config"][0]["dest_repo"].split("/")[-2]
+        dest_repo = RpmRepository.objects.get(pk=dest_repo_id)
+
+        source_repo_id = request.data["config"][0]["source_repo_version"].split("/")[-4]
+        source_repo_ver_number = request.data["config"][0]["source_repo_version"].split("/")[-2]
+
+        source_repo_version = RepositoryVersion.objects.get(
+            repository_id=source_repo_id, number=source_repo_ver_number
+        )
+        source_repo = RpmRepository.objects.get(pk=source_repo_version.repository_id)
+
+        if request.user.has_perm("rpm.change_rpmrepository", dest_repo) and request.user.has_perm(
```

Comment:
what the difference between change and modify repository permissions?
when i read those i take that change - allows to the change repo attributes, configuraiton,etc. modify - then relates to content modification. In that case I'd use here modify instead

Reply:
maybe worth calling it rpm.modify_content_rpmrepository